src/datamodel/validator/UMessage.cpp

```diff
@@ -339,6 +342,9 @@ ValidationResult isValidNotification(const v1::UMessage& umessage) {
 		if (!valid) {
 			return {false, Reason::BAD_SINK_URI};
 		}
+		if (umessage.attributes().sink().resource_id() != 0) {
+			return {false, Reason::BAD_SINK_URI};
+		}
```

Comment:
These checks should be in the URI validator. I would recommend splitting `isValidNotificaton()` into `isValidNotificationSource()` and `isValidNotificationSink()`. RPC already has separate source/sink validators (called request and response), and publish only uses one URI, so I think notifications are the only one that need the split.